functions/uploader/entities/ia-request.js

```diff
@@ -0,0 +1,97 @@
+const _ = require(`lodash`);
+const debug = require(`debug`)(`ia:uploader:entities:debug`);
+const error = require(`debug`)(`ia:uploader:entities:error`);
+const fetch = require(`node-fetch`);
+const mustache = require('mustache');
+
+const config = require('../../src/config');
+const {postEntitiesToDF, fetchEntitiesFromDF} = require('./entities');
+
+/**
+ * get unique & filtered entities for DialogFlow
+ *
+ * @param docs {array}
+ * @returns entities {array}
+ */
+function getUniqueCreatorsFromIA (docs) {
+  var creators = [];
+  var strCreator = ``;
+  for (let i = 0; i < docs.length; i++) {
+    var creator = docs[i].creator;
+    if (_.isArray(creator)) {
+      for (let i = 0; i < creator.length; i++) {
+        strCreator = creator[i];
+        if (strCreator) {
+          strCreator = strCreator.replace(/[()]+/g, ``);
+          creators.push(strCreator);
+        }
+      }
+    } else {
+      strCreator = creator;
+      if (strCreator) {
+        strCreator = strCreator.replace(/[()]+/g, ``);
+        creators.push(strCreator);
+      }
+    }
+  }
+  return _.uniq(creators);
+}
+
+/**
+ * get entities from InternetArchive
+ *
+ * @param id {string}
+ * @param limit {int}
+ * @returns entities {array}
+ */
+function fetchEntitiesFromIA (id, limit) {
+  var page = 0;
+  var sort = 'downloads+desc';
+  debug(`fetching entity data from IA...`);
+  var url = mustache.render(
+    config.endpoints.COLLECTION_ITEMS_URL,
+    {
+      id,
+      limit,
+      page,
+      sort,
+      fields: 'creator',
+    }
+  );
+  debug(url);
+  return fetch(url)
+    .then(res => res.json())
+    .then(data => {
+      debug(`fetched Entity from IA successfully.`);
+      return Promise.resolve(getUniqueCreatorsFromIA(data.response.docs));
+    }).catch(e => {
+      error(`Get error in fetching entity from IA, error: ${JSON.stringify(e)}`);
+      return Promise.reject(e);
+    });
+}
+
+/**
+ * get unique & filtered entities from InternetArchive & post to DialogFlow
+ *
+ * @param entityname {string}
+ * @param id {string}
+ * @param limit {int}
+ * @returns {promise}
+ */
+function fetchNewEntitiesFromIAAndPostToDF (entityname, id, limit) {
+  return Promise.all([
+    fetchEntitiesFromIA(id, limit),
```

Comment:
right now `fetchEntitiesFromIA` fetches only `creator`s but should fetch `genere`s as well

Reply:
@hyzhak : Actually I was considering creator to be the key for genres as well. then what should be the key for genres.

Reply:
@bimlendra  the field which you would like to fetch should be optional in `fetchEntitiesFromIA`

Reply:
@hyzhak : Definitely it should be but I was considering that creator is the field for genres & collection both. Please suggest me what is the name of field for genres?

Reply:
I am already checking duplicate values. Response of https://archive.org/advancedsearch.php?q=collection:(etree)&fl=subject&output=json is consisting "Live Concert" as well. Is it also a genre?

Reply:
@bimlendra Good question! Yes it makes sense, but you should consider the popular genres https://archive.org/advancedsearch.php?q=collection:(etree)&fl=subject&sort[]=downloads+desc&rows=200&page=0&output=json and postprocess them - split by `,` and if it is array split it by elements. 

Btw there is another API which could be used for fetching subjects https://archive.org/details/etree&morf=subject&headless=1&output=json which also has some duplications

Reply:
@hyzhak : But another api also giving "Grateful Dead" & "Live Concerts". It might confuse agent with other intents as it would mix Collections with Genres.

Reply:
@bimlendra  ok, so if we can't find the way how could we clean those genres for the moment we could use subjects of georgeblood only.

Reply:
Then I believe, we will have to put it on hold untill we get the solution.